### PR TITLE
fix: makes the persons modal work with the new export api

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

see #10607

The persons modal is disabled at the moment.

## Changes

This re-enables it using the new CSV exporter.

## How did you test this code?

adding developer tests and running locally and seeing CSVs download

### CODE REVIEW

**Comment by pauldambra** [comment]
@rcmarron @neilkakkar the person modal export was disabled in https://github.com/PostHog/posthog/pull/8877

I don't have much data locally so maybe I'm missing what is broken for it?

Should we leave it disabled?

File: frontend/src/scenes/trends/PersonsModal.tsx [hunk 1]
```diff
@@ -102,9 +103,11 @@ export function PersonsModal({
...
 
     const flaggedInsights = featureFlags[FEATURE_FLAGS.NEW_INSIGHT_COHORTS]
-    // TODO: Re-enable CSV downloads when frontend can support new entity properties
-    // const isDownloadCsvAvailable: boolean = view === InsightType.TRENDS && showModalActions && !!people?.action
-    const isDownloadCsvAvailable: boolean = false
+    const isDownloadCsvAvailable: boolean =
+        !!featureFlags[FEATURE_FLAGS.ASYNC_EXPORT_CSV_FOR_LIVE_EVENTS] &&
+        InsightType.TRENDS &&
+        showModalActions &&
+        !!people?.action
```
Comment (benjackwhite) [inline frontend/src/scenes/trends/PersonsModal.tsx:110]: Generally I say 👍  I wonder though if we couldn't in the future abstract this modal much more nicely to take some sort of `url` parameter which can then automagically be used to load the data relevant to the modal / pagination etc. Would make the CSV exporting much simpler as well then as it simply uses the provided url...

**Review by benjackwhite — APPROVED** [review]
For now looks great, small rambling thought about how we might be able to do things more nicely in the future

**Comment by neilkakkar** [comment]
Hmm, I think it's still a mess, not fixed enough to enable.

Here's some specific cases where its borked:

1. When there's a filter on the trend entity, and a global filter
2. The above^, with breakdowns involved

image


It doesn't seem to crash at all anymore, but I suspect the results would be inconsistent / wrong, given that the URL generated is kinda wrong, example:

```
/api/projects/1/actions/people?insight=TRENDS&events=
[{"id":"$pageview","name":"$pageview","type":"events","order":0,"properties":[{"key":"$browser","type":"event","value":["Chrome"],"operator":"exact"}]}]&actions=[]&display=ActionsLineGraph&interval=day&new_entity=[]
// the wrong bit, parsing an array as dict, or vice versa
&properties={"0":{"key":"$browser","value":["Chrome"],"operator":"exact","type":"person"},"1":{"key":"$browser","value":["Chrome"],"operator":"exact","type":"event"},"type":"AND","values":[{"key":"$browser","operator":"exact","type":"event","value":["Chrome"]}]}
// ^^^^^^
&filter_test_accounts=false&date_from=2022-07-18&entity_id=$pageview&entity_type=events&breakdown_value=&date_to=2022-07-18&label=$pageview"
```

**Comment by neilkakkar** [comment]
Although, if you're working on this, happy to help figure out fixes! (Don't mind getting this in as is, since feature flagged, and we can iron out the rough edges. Prod data should give a better sense of other cases where its borked.

**Comment by pauldambra** [comment]
> Although, if you're working on this, happy to help figure out fixes! 

I think I'll swerve and avoid this pothole 🤣 

I'll get this in behind two flags so we can test for ourselves but we're not coupled to release of CSV exports

**Pull request merged**